unstructured/ingest/connector/jira.py

```diff
@@ -0,0 +1,491 @@
+            kwargs["start"] += kwargs["limit"]
+
+        return all_results[:number_of_items_to_fetch]
+
+    return wrapper
+
+
+@dataclass
+class JiraIngestDoc(IngestDocSessionHandleMixin, IngestDocCleanupMixin, BaseIngestDoc):
+    """Class encapsulating fetching a doc and writing processed results (but not
+    doing the processing).
+
+    Current implementation creates a Jira connection object
+    to fetch each doc, rather than creating a it for each thread.
+    """
+
+    config: SimpleJiraConfig
+    file_meta: JiraFileMeta
+    registry_name: str = "jira"
+
+    @property
+    def source_url(self):
+        return f"{self.config.url}/browse/{self.file_meta.issue_key}"
+
+    @property
+    def record_locator(self):  # Values must be JSON-serializable
+        """A dictionary with any data necessary to uniquely identify the document on
+        the source system."""
+        return {"issue_key": self.file_meta.issue_key}
+
+    @requires_dependencies(dependencies=["atlassian"], extras="jira")
+    def get_metadata_fields(self):
+        try:
+            jira = self.session_handle.service
+            issue = jira.issue(self.file_meta.issue_key)
+            issue_fields = nested_object_to_field_getter(issue["fields"])
+
+            self.metadata_fields = {
+                "date_modified": str(issue_fields["updated"]),
+                "date_created": str(issue_fields["created"]),
+                "date_processed": str(datetime.datetime.now().time()),
+                "record_locator": self.record_locator,
+            }
+
+            self.check_exists = True
+            self.file_exists = True
+
+        except HTTPError as error:
+            if error.response.reason == "Not Found":
+                self.check_exists = True
+                self.file_exists = False
+            else:
+                logger.error(f"Error: {error} for issue {self.file_meta.issue_key}")
+
+    @property
+    def date_created(self) -> Optional[str]:
+        if not hasattr(self, "metadata_fields"):
+            self.get_metadata_fields()
+
+        return self.metadata_fields["date_created"] if hasattr(self, "metadata_fields") else None
+
+    @property
+    def date_modified(self) -> Optional[str]:
+        if not hasattr(self, "metadata_fields"):
+            self.get_metadata_fields()
+
+        return self.metadata_fields["date_modified"] if hasattr(self, "metadata_fields") else None
+
+    @property
+    def date_processed(self) -> Optional[str]:
+        if not hasattr(self, "metadata_fields"):
+            self.get_metadata_fields()
+
+        return self.metadata_fields["date_processed"] if hasattr(self, "metadata_fields") else None
+
+    @property
+    def exists(self) -> Optional[bool]:
+        """Whether the document exists on the remote source."""
+        if self.check_exists:
+            return self.file_exists
+
+        self.get_metadata_fields()
+
+        return self.file_exists
```

Comment:
why not just make `exists` a field in `metadata_fields` that way you can just `return self.metadata_fields.exists`?

Reply:
If this means a dict value, addressed with https://github.com/Unstructured-IO/unstructured/pull/1238/commits/4224f0352ee720a80d9cf3574fb48cd73082ca1d, if not please let me know

Reply:
ahh, yep, just wasn't seeing that in the UI, but that's perfect